connector/src/main/java/org/geysermc/connector/network/session/GeyserSession.java

```diff
@@ -301,6 +302,12 @@
     @Setter
     private boolean waitingForStatistics = false;
 
+    /**
+     * Stores advancements for the player.
+     */
+    private final Map<String, Advancement> storedAdvancements = new HashMap<>();
+    @Setter
+    private Map<Integer, String[]> buttonIdsToIdAndTitleAdvancements = new HashMap<>();
```

Comment:
There should be a gap before and after this, along with a JavaDoc comment explaining it. There also may be a better way to do this instead of having another session variable, maybe a custom class and combine it with the stored advancements?